### PR TITLE
junbon/multiplier-main

### PR DESCRIPTION
**Pull request by junbon-binary**

### CODE REVIEW

File: share/longcodes.yml [hunk 1]
```diff
@@ -80,4 +80,6 @@ putspread_intraday_fixed_expiry: "Win up to [_7] [_6] if [_1]'s exit tick is bet
 putspread_daily: "Win up to [_7] [_6] if [_1]'s exit tick is between [_4] and [_5] at [_3]."
 runhigh_tick: "Win payout if [_1] rises successively for [plural,_3,%d tick, %d ticks] after the entry spot."
 runlow_tick: "Win payout if [_1] falls successively for [plural,_3,%d tick, %d ticks] after the entry spot."
+multup_daily: "longcode for multup"
+multdown_daily: "longcode for multdown"
```
Comment (dmitry-binary) [inline share/longcodes.yml]: The longcode description for these two diverges a bit from the rest. Do these intend to be displayed in UI? Perhaps, better to ask help from copywriters?

File: lib/Finance/Contract/Longcode.pm [hunk 2]
```diff
@@ -184,9 +188,16 @@ sub shortcode_to_parameters {
 
     return $legacy_params if (not exists Finance::Contract::Category::get_all_contract_types()->{$initial_bet_type} or $shortcode =~ /_\d+H\d+/);
 
-    if ($shortcode =~
+    if ($shortcode =~ /^(MULTUP|MULTDOWN)_(R?_?[^_\W]+)_(\d*\.?\d*)_(\d+)_(\d+)_(\d+)$/) {
```
Comment (ravi-binary) [inline lib/Finance/Contract/Longcode.pm]: Will **(R?_?[^_\W]+)** work for all underlyings or will we have to change this when we launch for Fx or  fast indices?

File: share/longcodes.yml (hunk 1 above)
Comment (junbon-binary) [reply share/longcodes.yml]: These are just made up. No one has time to think about longcode just yet.

File: lib/Finance/Contract/Longcode.pm (hunk 2 above)
Comment (junbon-binary) [reply lib/Finance/Contract/Longcode.pm]: works for everything.

**Pull request closed without merging**